components/wifi/esp8266-driver/ESP8266/ESP8266.h

```diff
@@ -428,7 +428,6 @@ class ESP8266 {
     PinName _serial_rts;
     PinName _serial_cts;
     rtos::Mutex _smutex; // Protect serial port access
-    rtos::Mutex _rmutex; // Reset protection
```

Comment:
Why is `_rmutex` removed while `_smutex` can stay? I thought that "bare-metal" means "no RTOS", means thread synchronisation mechanisms?

Reply:
I helped prepare this.

`_rmutex` wasn't doing anything, as far as I can see. It was a remnant of the also-redundant condition variable bits for reset, which seemed to be imagining some other thread running OOB handlers.

The piece of code waiting for reset is inside `_smutex`, so it all happens synchronously, blocking inside that code.